### ISSUE TITLE
Update stac_ipyleaflet to use titiler-pgstac endpoints

### ISSUE THREAD

**Issue by hrodmn** [MAAP-Project/stac_ipyleaflet#131]
### Description
We are working on consolidating the titiler deployments in MAAP to a single instance (titiler-pgstac.maap-project.org) (https://github.com/NASA-IMPACT/active-maap-sprint/issues/945, https://github.com/NASA-IMPACT/active-maap-sprint/issues/988, https://github.com/NASA-IMPACT/active-maap-sprint/issues/471).

I will go through stac-ipyleaflet and figure out which (if any) of the features will require any custom titiler modifications (e.g. `mosaicjson` endpoints) or if we can use new features in titiler-pgstac to replace those custom features.

### Functions that use titiler-stac.maap-project.org:
- [x] `Stac.get_tile_url`: will work if we add the `/cog` endpoints
- [ ] `Stac.stac_tile`
  - Need to use `/collections/{collection_id}/items/{item_id}/{tileMatrixSetId}/tilejson.json` instead of `/stac/{TileMatrixSetId}/tilejson.json`
  - OR add `/stac` endpoints to the titiler-pgstac tiler
- [ ] `Stac.get_metadata`: No more `/metadata` endpoint... but I can't tell if this function is actually used anywhere. Still need to figure out what is returned by this feature.

cc @abarciauskas-bgse @jjfrench

**Comment by hrodmn** [comment]
@jjfrench if we add the `/stac` endpoints to `titiler-pgstac.maap-project.org` then we would not need to update anything in `stac_ipyleaflet` except for the URL for `TITILER_STAC_ENDPOINT`. We could update `stac_ipyleaflet` to use the `/collections/{collection}/{item}` endpoints from `titiler-pgstac` but that would assume that the `TITILER_STAC_ENDPOINT` is running `titiler-pgstac` (not just `titiler`). I am not familiar with how this package gets used in other places so it might be simpler to make `titiler-pgstac.maap-project.org` compatible with `stac_ipyleaflet` instead of the other way around. What do you think?

**Comment by jjfrench** [comment]
@hrodmn I think that would be reasonable, but I also think that this package isn't being used i.e other packages are more popular or stac-ipyleaflet is missing out on key features. I'm unsure about the longevity of this package as we haven't supported development on it for quite awhile now

**Comment by jjfrench** [comment]
@wildintellect thoughts on this?

**Comment by wildintellect** [comment]
I think finding the mosiacjson usage should be easy, pretty sure it's all on the mosaics tab, but should also be easy to swap for titiler-pgstac mosaics instead. The bigger challenge is that dynamic mosaics are not really supported in stac_ipyleafet yet (no gui for it). So the question for the long term roadmap is what's the next evolution of this tool given `Lonboard` and aligning with VEDA. 

@jjfrench can you confirm, with this upgrade did we break the existing stac_ipyleaflet version? or will that happen only once we replace the main titiler instance?

Thoughts
@kylebarron @sandrahoang686 @j08lue

**Comment by jjfrench** [comment]
@wildintellect, it would only break the current version once we replace/remove the titiler-stac instance - as of right now, titiler-stac is still deployed

**Comment by vincentsarago** [comment]
> OR add /stac endpoints to the titiler-pgstac tile

FYI I'm not planning to add such endpoint to titiler-pgstac module but this could be done in a custom app.

>  I am not familiar with how this package gets used in other places so it might be simpler to make titiler-pgstac.maap-project.org compatible with stac_ipyleaflet instead of the other way around. What do you think?

I would kinda agrees, if we think stac-ipyleaflet could be use by any stac endpoint then it should at least works with `common` titiler deployment that has the `/stac` endpoint.

as I said I'm not planning to add this `/stac` endpoint to titiler-pgstac because it doesn't really fit with the purpose of the package. it's easy enough for people to add endpoints on their own deployment

**Comment by hrodmn** [comment]
@vincentsarago yes I agree that the `/stac` endpoints are an odd thing to add to a `titiler-pgstac` deployment since it is so easy to get tiles for items and collections with the `/collections` endpoints. As you say, it is simple to add to `/stac` endpoints to the existing custom titiler-pgstac deployment! https://github.com/MAAP-Project/maap-eoapi/pull/27

**Comment by j08lue** [comment]
> the question for the long term roadmap is what's the next evolution of this tool given Lonboard and aligning with VEDA

We have an overall goal to make VEDA JupyterHub equivalent to MAAP ADE. If stac_ipyleaflet is a popular feature of MAAP ADE, we should keep maintaining it, also under VEDA, so users who change between the platforms can use the same tool both places.

Otherwise, we currently on VEDA do not have a strong use case for further developing stac_ipyleaflet, I would say. Happy to be convinced otherwise.

Lonboard has some unique capabilities for interactive vector data exploration that we would like to advertise for on VEDA. Once that advertisement is complete, also there I would say we should not plan for major development on VEDA, unless prominent VEDA users request new features.

In short, in terms of user value:

* stac_ipyleaflet: _discover and render_ raster datasets from curated STAC catalogs provided by the platform, in a notebook environment
* Lonboard: explore large vector datasets interactively with rapid response and on-the-fly filtering, in a notebook environment, with support for adding selected tiled raster datasets to the same map

I think where the two differ is that Lonboard is more like a power-ipyleaflet, while stac_ipyleaflet's value comes from the convenient connection to STAC. I guess there are no plans to make Lonboard STAC-aware.

**Comment by kylebarron** [comment]
> I guess there are no plans to make Lonboard STAC-aware

Not particularly. But there's potential to make an extension to Lonboard to support STAC
